dot/network/host.go

```diff
@@ -27,31 +27,43 @@ import (
 	ma "github.com/multiformats/go-multiaddr"
 )
 
-var publicIP string = ""
-
-var privateCIDRs = []string{
-	"10.0.0.0/8",
-	"172.16.0.0/12",
-	"192.168.0.0/16",
-	"100.64.0.0/10",
-	"198.18.0.0/15",
-	"169.254.0.0/16",
+func newPrivateIPFilters() (privateIPs *ma.Filters, err error) {
+	privateCIDRs := []string{
+		"10.0.0.0/8",
+		"172.16.0.0/12",
+		"192.168.0.0/16",
+		"100.64.0.0/10",
+		"198.18.0.0/15",
+		"169.254.0.0/16",
+	}
+	privateIPs = ma.NewFilters()
+	for _, cidr := range privateCIDRs {
+		_, ipnet, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return privateIPs, err
+		}
+		privateIPs.AddFilter(*ipnet, ma.ActionDeny)
+	}
+	return
+}
+
+var (
+	privateIPs *ma.Filters
+)
+
+func init() {
+	var err error
+	privateIPs, err = newPrivateIPFilters()
+	if err != nil {
+		log.Panic(err)
+	}
```

Comment:
Please remove this and use 

```go
	return ip.IsPrivate() || ip.IsLoopback() ||
		ip.IsLinkLocalUnicast() || ip.IsLinkLocalMulticast()
```

to check if an IP is private. This is hugely faster, require no dependency and works with IPv4, IPv6 and IPv4 wrapped in IPv6.

Alternatively add missing private networks to your list.